pkg/controller/usercleanup/user_cleanup_controller_test.go

```diff
@@ -0,0 +1,132 @@
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"testing"
+	"time"
+
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+func TestUserCleanup(t *testing.T) {
+	t.Run("test that user cleanup doesn't delete an active UserSignup", func(t *testing.T) {
+		userSignup := &v1alpha1.UserSignup{
+			ObjectMeta: test2.NewUserSignupObjectMeta("", "abigail.thompson@redhat.com"),
+			Spec: v1alpha1.UserSignupSpec{
+				UserID:      "User-98887",
+				Username:    "abigail.thompson@redhat.com",
+				Deactivated: false,
+			},
+			Status: v1alpha1.UserSignupStatus{
+				Conditions: []v1alpha1.Condition{
+					{
+						Type:   v1alpha1.UserSignupComplete,
+						Status: v1.ConditionTrue,
+					},
+					{
+						Type:   v1alpha1.UserSignupApproved,
+						Status: v1.ConditionTrue,
+						Reason: "ApprovedAutomatically",
+					},
+				},
+				CompliantUsername: "abigail-thompson",
+			},
+		}
+		userSignup.Labels["toolchain.dev.openshift.com/approved"] = "true"
+		userSignup.Labels[v1alpha1.UserSignupStateLabelKey] = "approved"
+
+		r, req, _ := prepareReconcile(t, userSignup.Name, userSignup)
+
+		_, err := r.Reconcile(req)
+		require.NoError(t, err)
+
+		// Confirm the UserSignup still exists
+		key := test.NamespacedName(test.HostOperatorNs, userSignup.Name)
+		require.NoError(t, r.client.Get(context.Background(), key, userSignup))
+		require.NotNil(t, userSignup)
+	})
+
+	t.Run("test that user cleanup doesn't delete a recently deactivated UserSignup", func(t *testing.T) {
+
+		userSignup := &v1alpha1.UserSignup{
+			ObjectMeta: test2.NewUserSignupObjectMeta("", "brian.anderson@redhat.com"),
+			Spec: v1alpha1.UserSignupSpec{
+				UserID:      "User-93321",
+				Username:    "brian.anderson@redhat.com",
+				Approved:    true,
+				Deactivated: true,
+			},
+			Status: v1alpha1.UserSignupStatus{
+				Conditions: []v1alpha1.Condition{
+					{
+						Type:               v1alpha1.UserSignupComplete,
+						Status:             v1.ConditionTrue,
+						Reason:             v1alpha1.UserSignupUserDeactivatedReason,
+						LastTransitionTime: metav1.Time{Time: time.Now()},
+					},
+					{
+						Type:   v1alpha1.UserSignupApproved,
+						Status: v1.ConditionTrue,
+						Reason: "ApprovedAutomatically",
+					},
+				},
+				CompliantUsername: "brian.anderson",
+			},
+		}
+		userSignup.Labels["toolchain.dev.openshift.com/approved"] = "true"
+		userSignup.Labels[v1alpha1.UserSignupStateLabelKey] = "approved"
+
+		r, req, _ := prepareReconcile(t, userSignup.Name, userSignup)
+
+		_, err := r.Reconcile(req)
+		require.NoError(t, err)
+
+		// Confirm the UserSignup still exists
+		key := test.NamespacedName(test.HostOperatorNs, userSignup.Name)
+		require.NoError(t, r.client.Get(context.Background(), key, userSignup))
+		require.NotNil(t, userSignup)
```

Comment:
Could you please verify that the `Requeue` is `true` and the `RequeueAfter` is approx within the range of the expected time?

Reply:
Done.  I've added a check here to confirm the `RequeueAfter` is within a 2 day range of the default expected value.  We might want to tighten this up though, what do you think?